src/draw/vg_lite/lv_draw_vg_lite.c

```diff
@@ -134,21 +135,15 @@ static void draw_execute(lv_draw_vg_lite_unit_t * u)
             break;
     }
 
-#if LV_USE_PARALLEL_DRAW_DEBUG
-    /* Layers manage it for themselves. */
-    if(t->type != LV_DRAW_TASK_TYPE_LAYER) {
-    }
-#endif
-
-    LV_VG_LITE_CHECK_ERROR(vg_lite_finish());
+    lv_vg_lite_flush(draw_unit);
```

Comment:
Calling vg_lite_flush() after each task does not necessary mean that command gets batched. Instead, after each vg_lite_flush() you will force the other command buffer to take over. As GPU have only 2 command buffers, if a 3th vg_lite_flush() comes too fast while the first vg_lite_flush() is not yet complete - e.g. the command buffer 1 is not complete - then the 3'th vg_lite_flush() will act as a vg_lite_finish().

This mechanism does not work as intend to.

What is the improvement with this new code?

Reply:
Remember that vg_lite_flush() will switch the two GPU command buffers.

Reply:
This function is a double encapsulation of `vg_lite_flush`.
In `lv_vg_lite_flush` I added the count. When `lv_vg_lite_flush` is called more than a certain number of times, `vg_lite_flush` is called to tell the GPU to start rendering, so that the GPU can process several small tasks together to improve efficiency.

Reply:
I understand now.

But you batch by force a number of LV_VG_LITE_FLUSH_MAX_COUNT tasks! Even the first time.

In my tests I found that GPU gets busy for a couple of 4-5 tasks only when it has to deal with filling the screen area - big task. Otherwise the GPU completes any task right away.

So the GPU gets IDLE most of the time and you do not have to batch anymore commands. This is why I suggest the solution I implemented here:
https://github.com/lvgl/lvgl/blob/3199231b39b267c91733225a1e521d2784ee74a8/src/draw/nxp/vglite/lv_vglite_utils.c#L80
In this way I am making sure the GPU is busy as much as possible. If the GPU is idle then makes no sense to queue even more commands.

Reply:
What if we implement this logic inside `vg_lite_flush`? Let the driver determine the GPU status by itself. If the GPU is busy, it will join the queue. If it is idle, it will start rendering immediately.

Reply:
That is an API change behavior. In documentation it says it "submits the command to the GPU without waiting for completion". You can not just return because it might not be come another vg_lite_flush() to really submit it to the GPU.

There are other applications beside lvgl on top of vglite or in some cases customers can use vglite api directly to design the application. I would not ask for a change.

Reply:
Hmm...that makes sense. May I ask if you added the `VG_LITE_GPU_IDLE_STATE` enumeration in `vglite_run` yourself? Or is it already implemented in the SDK?

Reply:
We asked for that implementation to the VSI. You should see it as well in the next driver releases. I use 4.0.58 version.

Reply:
Great! I'll do the same thing.

Reply:
I would also let the VGLite thread implementation with the possibility to run it single thread (v8 like).
https://github.com/lvgl/lvgl/blob/152dc0b2beafa12e9030dc614d5bc3535e1e9c6a/src/draw/nxp/vglite/lv_draw_vglite.c#L113
And check all the places where the ```#if LV_USE_OS``` is used in this file. Simply setting the LV_USE_OS to LV_OS_NONE will make it run single thread (as you have it now).